### PR TITLE
Families for enterprise/add sponsorship prevalidate

### PR DESCRIPTION
**Pull request by MGibson1**
## Type of change
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective
Add sponsorship pre-validate endpoint which returns whether the provided token is valid or not.

## Testing requirements
None, test via web PR

## Before you submit
- [ ] If making database changes - I have also updated Entity Framework queries and/or migrations
- [x] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

**Comment by MGibson1** [comment]
looking into failures

File: src/Core/Services/IOrganizationSponsorshipService.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@ namespace Bit.Core.Services
 {
     public interface IOrganizationSponsorshipService
     {
-        Task<bool> ValidateRedemptionTokenAsync(string encryptedToken);
+        Task<bool> ValidateRedemptionTokenAsync(string encryptedToken, string currentUserEmail);
```
Comment (Hinton) [inline src/Core/Services/IOrganizationSponsorshipService.cs:10]: We seem to mostly be using `sponsoring` / `sponsored` concept here. `currentUserEmail` feels a bit out of place. Maybe `sponsoredUserEmail`?

**Review by Hinton — DISMISSED** [review]

File: src/Api/Controllers/OrganizationSponsorshipsController.cs [hunk 2]
```diff
@@ -67,11 +65,18 @@ await _organizationSponsorshipRepository
                 (await CurrentUser).Email);
         }
 
+        [HttpPost("validate-token")]
+        [SelfHosted(NotSelfHostedOnly = true)]
+        public async Task<bool> PreValidateSponsorshipToken([FromQuery] string sponsorshipToken)
```
Comment (Hinton) [inline src/Api/Controllers/OrganizationSponsorshipsController.cs:70]: We don't usually use query params in combination with post. Any particular reason?

Comment (MGibson1) [reply src/Api/Controllers/OrganizationSponsorshipsController.cs:70]: I wanted to match the `redeem` endpoint. Now as to why that one has a query parameter, the best I can remember is I wanted a symmetry between the link we give in the email offer and the endpoint, but that feels quite flimsy now...

**Review by Hinton — APPROVED** [review]

File: src/Api/Controllers/OrganizationSponsorshipsController.cs (hunk 2 above)
Comment (Hinton) [reply src/Api/Controllers/OrganizationSponsorshipsController.cs:70]: Fair enough.

**Pull request merged**